app/objects/beatmap.py

```diff
@@ -588,35 +588,18 @@ def url(self) -> str:  # same as above, just no beatmap id
         """The online url for this beatmap set."""
         return f"https://osu.{app.settings.DOMAIN}/beatmapsets/{self.id}"
 
-    def all_officially_ranked_or_approved_or_frozen(self) -> bool:
-        """Whether all the maps in the set are
-        ranked or approved on official servers."""
-        return all(
-            # ranked status has been edited on bancho.py
-            bmap.frozen or
-            # ranked status is ranked or approved on bancho
-            bmap.status in (RankedStatus.Ranked, RankedStatus.Approved)
-            for bmap in self.maps
-        )
-
-    def all_officially_loved_or_frozen(self) -> bool:
-        """Whether all the maps in the set are
-        loved on official servers."""
-        return all(
-            # ranked status has been edited on bancho.py
-            bmap.frozen or
-            # ranked status is loved on bancho
-            bmap.status == RankedStatus.Loved
-            for bmap in self.maps
+    def any_beatmaps_have_official_leaderboards(self) -> bool:
+        """Whether all the maps in the set have leaderboards on official servers."""
+        leaderboard_having_statuses = (
+            RankedStatus.Loved,
+            RankedStatus.Ranked,
+            RankedStatus.Approved,
```

Comment:
```suggestion
            RankedStatus.Approved,
            RankedStatus.Qualified,
```

Reply:
qualified queue is generally quite slow on updates (~7 days after qualification) so it's safe & likely more efficient to treat it the same as the rest

Reply:
i think they're different enough to warrant different cases to avoid the case of incorrectly high confidence near the end of the 7d window